nssp/delphi_nssp/patch.py

```diff
@@ -0,0 +1,195 @@
+"""
+This module is used for patching data in the delphi_nssp package.
+
+To use this module, you need to turn on the custom_run flag
+and specify the range of issue dates in params.json, like so:
+
+{
+  "common": {
+    "custom_run": true,
+    ...
+  },
+  "validation": {
+    ...
+  },
+  "patch": {
+    "source_backup_credentials": {
+      "host": "bigchunk-dev-02.delphi.cmu.edu",
+      "user": "user",
+      "path": "/common/source_backup/nssp"
+    },
+    "patch_dir": "/Users/minhkhuele/Desktop/delphi/covidcast-indicators/nssp/AprilPatch",
+    "start_issue": "2024-04-20",
+    "end_issue": "2024-04-21",
+    "source_dir": "/Users/minhkhuele/Desktop/delphi/covidcast-indicators/nssp/source_data"
+  }
+}
+
+It will generate data for that range of issue dates, and store them in batch issue format:
+[name-of-patch]/issue_[issue-date]/nssp/actual_data_file.csv
+"""
+
+import sys
+from datetime import datetime, timedelta
+from os import makedirs, path
+
+import pandas as pd
+import paramiko
+from delphi_utils import get_structured_logger, read_params
+from epiweeks import Week
+
+from .run import run_module
+
+
+def get_source_data(params, logger):
```

Comment:
I'm not quite sure what's the purpose of this function here. And furthermore, it doesn't seem like it's cleaned up at the end. Do we need to need to keep the csv?

Reply:
yup. It's intentional to keep the csv. 
The purpose of the function is to automate grabbing source data. Before adding this, whoever does patching has to scp the source files into source_dir before running patch.py.

Reply:
Hm... How often do we have to re run patches like so? I understand it's time consuming to pull, but I'm not quite convinced that we would need to keep them after the run. I feel like it's going to get cluttered and messy.

Reply:
That's super fair. Imma add something to delete them after each run then.

Reply:
added.